### ISSUE TITLE
Implement torch.py for APS

### ISSUE THREAD

**Issue by zelihabeyza** [Cortys/SEP-Probly-WS2526#326]
### **Goals**
- Implement `torch.py`

**Comment by zelihabeyza** [comment]
Implemented `torch.py`:
- **implements APS for PyTorch neural networks**
- Initialization `__init__()`: takes PyTorch model, handles device management, puts model in evaluation mode
- Probability Extraction `_get_probs()`: converts input to PyTorch tensor, runs model inference, converts outputs to probabilities, returns numby array for combatibility with APS calculations
- Non-Conformity Scoring `_compute_nnconformity()`: gets predicted probabilities from model and uses `calculate_nonconformity_score()` from conformal_prediction/aps/common.py
- Calibration `calibrate()`: determines how strict the prediction sets should be
- Prediction `predict()`: If calibrated with significance=0.1, true label will be in set 90% 

`torch.py`: https://github.com/m3ryemmo/probly/blob/feature/conformal-prediction-aps/src/probly/conformal_prediction/aps/torch.py
